Elastic indices 3 (#484)
ElrondNetwork/docs.elrond.com: docs/sdk-and-tools/indices/accountsesdt.md

---
id: es-index-accountsesdt
title: Elasticsearch index: accountsesdt
---

The `_id` field of this index is composed in this way: `{bech32address}_{tokenIdentifier}_{nonce}` (example: `erd.._abcd-0123-01`).

| Field       | Description                                                                                                                           |
|-------------|---------------------------------------------------------------------------------------------------------------------------------------|

> | identifier  | The identifier field is composed from the `token` field and the `nonce` field hex encoded.                                            |

Reviewer: ```suggestion
| identifier  | The identifier field consists of `token` field and the `nonce` field hex encoded.                                            |
```
Also, could you add an example here?

Reviewer: I think an example would be good here. How exactly are `token` and `nonce` merged for example ("ALICE-nonce"/ "ALICEnonce")?

Author: added an example

| address     | The address field holds the address in a bech32 encoding.                                                                             |

> | balance     | The balance field holds the amount of ESDT token the address possesses. It is a string that also includes the number of decimals.     |

Author: changed.

> | balance     | The balance field holds the amount of ESDT token the address possesses. It is a string that also includes the number of decimals.     |

Reviewer 2: not done yet. you just removed the sentence

> | balance     | The balance field holds the amount of ESDT token the address possesses. It is a string that also includes the number of decimals.     |

Author: changed.

> | balanceNum  | The balanceNum field holds the amount of ESDT token the address possesses, in a numeric format.                                       |

Author: done

| data        | The data field is a structure that contains extra data about a token, such as the creator of an NFT.                                  |

> | tokenNonce  | The tokenNonce field holds the sequence number of the token. This field can be empty in the case of `FungibleESDT`.                   |

Author: changed

> | token       | The token field holds the token name of the token.                                                                                    |

Author: done

| timestamp   | The timestamp field represents the timestamp when the address balance was changed.                                                    |
| type        | The type field represents the type of the ESDT token. It can be `FungibleESDT`, `NonFungibleESDT`, `SemiFungibleESDT`, or `MetaESDT`. |

> The docs with the field `tokenNonce` non-empty can have the `data` field populated with the fields below:

Reviewer: Docs? What do you mean by that? Is it some elastic indexer specific term?

```suggestion
Docs with a non-empty `tokenNonce` field will have the following `data` fields:
```

Reviewer: What is docs?

| data fields        | Description                                                                     |
|--------------------|---------------------------------------------------------------------------------|

> | uris               | The uris field holds a URL or a web address.                                    |

Reviewer: ```suggestion
| uris               | The uris field holds an URL or a web address.                                    |
```

Author: done

> | uris               | The uris field holds a URL or a web address.                                    |

Author: changed

> | creator            | The creator field holds the address bech32 encoded of the creator of the token. |

Reviewer: ```suggestion
| creator            | The creator field holds the bech32 encoded address of the creator of the token. |
```

Author: done.

> | creator            | The creator field holds the address bech32 encoded of the creator of the token. |

Author: changed.

> | whiteListedStorage | The whiteListedStorage field is true if the token has white-listed storage.     |

Reviewer 2: not done yet

| attributes         | The attributes field contains the attributes of the token.                      |

> | nonEmptyURIs       | The nonEmptyURIs field is true if the token has nonce empty uris.               |

Reviewer: Isn't it actually the other way around? (judging by the name `nonEmptyURIs`)

True if **NOT** empty

Author: changed.

> | nonEmptyURIs       | The nonEmptyURIs field is true if the token has nonce empty uris.               |

Author: changed.

## Query examples

> ### Fetch addresses sorted by balance with a specific token

Author: changed

```
curl --request GET \
  --url ${ES_URL}/accountsesdt/_search \
  --header 'Content-Type: application/json' \
  --data '{
    "query": {
        "match": {
         "identifier": {
           "query": "MY-TOKEN-aaabbb",
           "operator": "AND"
         }
       }
     }
    "sort": [
        {
            "balanceNum": {
                "order": "desc"
            }
        }
    ],
    "size":10
}'
```

> ### Fetch all the ESDT tokens that belong to an address.

Author: changed.
